global/scripts/parallel.shlib

```diff
@@ -159,6 +162,10 @@ function par_numlog()
 
 par_jobarray=()
 par_jobinputarray=()
+
+par_fslsub_joblist=()
+par_fslsub_job_waitlist=()
+
```

Comment:
As discussed, we don't really want any fsl_sub stuff inside a pipeline script (can't work inside qunex at all, makes things complicated because of job dependencies that may need to escape a particular script and be used by another, and no consistent way to otherwise synchronize jobs), and the .shlibs are (so far) only intended for use inside pipelines.  Something is either a launcher batch script (and can use fsl_sub), or a pipeline script (and doesn't use fsl_sub, but can use backgrounding, which parallel.shlib is intended to simplify control of).

My inclination is to ask that you use the existing `parallel.shlib` without adding any fsl_sub stuff to it (revert it to master, not sure if even the edit adding the "no jobs" warning is particularly useful, I can imagine situations where it is simpler to not keep track of whether a loop actually resulted in any added jobs, and yet wouldn't deserve a warning).

It isn't clear to me that `parallel.shlib` functions should be involved at all in non-parallel mode, either.  What exactly is the use case of non-parallel mode?

Reply:
Please see "Parallel modes" in the updated annotation where I try to address these questions.

Reply:
Where is the "updated annotation" you are talking about?  I don't see any justification mentioned in the help info for the `--parallel-modes` options, or in the edits to `parallel.shlib`.  Where should I be looking?

Previously, we had a clean separation between batch launchers that can use `fsl_sub` (because they don't run any processing themselves, they just tell the cluster to run other scripts, which are usually independent so we don't even need job dependencies), and processing pipelines that can't (because `fsl_sub` can't play nice when intermixed with simple serial command sequences - if you submit a job, and you need to run something after it finishes, that would need to become a job with a dependency, no matter how simple it is - and then the command after it would need to be a job with a dependency, and the next...).  `parallel.shlib` was designed as a more sophisticated way to control shell backgrounding, which allowed you to go back to simple, literal serial commands when you were done with a section of parallel stuff.  All the shlibs are meant to be used only in pipeline scripts, not launchers, hence, `fsl_sub` wasn't on the table for `parallel.shlib`.

What is the justification for making `parallel.shlib` more complicated, compared to keeping the fsl_sub complications contained to the batch launcher scripts?  Why should there even be a non-parallel mode for a pipeline script that uses parallel.shlib?  Complication without a clear use case doesn't make sense.